swap_meet/vendor.py

```diff
@@ -1,2 +1,90 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        
+        self.inventory = inventory if inventory is not None else []
+
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+
+    def get_by_category(self, category):
+
+        # filter method searches through the inventory and returns a list if the item matches the category
+        category_items = list(filter(lambda x: x.category == category, self.inventory))
+
+        # alternative method is using list comprehension
+        # category_items = [item for item in self.inventory if item.category == category]
+
+        # without filter or list comprehension
+        # category_items = []
+        # for item in self.inventory:
+        #     if item.category == category:
+        #         category_items.append(item)
+
+        return category_items
+
+
+    def swap_items(self, vendor, my_item, their_item):
+
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
```

Comment:
Nice guard clause to make sure that the specified items are present in the lists before trying to swap them.